.justfile

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env just --justfile
+
+set dotenv-load := true
+
+repo := `git rev-parse --show-toplevel`
+profile := env_var_or_default('PROFILE', "dev")
+github-event-name := env_var_or_default('GITHUB_EVENT_NAME', "none")
+
+
+######################################################################
+## Helper to print a message when calling `just`
+######################################################################
+[private]
+@default:
+  echo "Usage: just <recipe>"
+  just list-repo-recipes
+  echo "For further information, run 'just --help'"
+
+# List recipes in this file and from the calling directory
+[no-cd]
+[private]
+@usage:
+  echo "Usage: just <recipe>"
+  just list-local-recipes
+  just list-repo-recipes
+  echo "For further information, run 'just --help'"
+
+[no-cd]
+[private]
+@list-local-recipes:
+  echo '\nAvailable recipes:'
+  just --list --unsorted --list-heading ''
+
+[private]
+@list-repo-recipes:
+  echo "\nRepository recipes:"
+  just --list --unsorted --list-heading ''
+
+
+######################################################################
+## Helper to run a global command
+######################################################################
+
+# Runs yarn in the repository root
+[private]
+yarn *args:
+  yarn {{args}}
+
+
+######################################################################
+## Helper to run a command on an environmental condition
+######################################################################
+
+# Runs the provided command if `PROFILE` starts with `"dev"`
+[private]
+[no-cd]
+in-dev +command:
+  #!/usr/bin/env bash
+  set -euo pipefail
+  if [[ {{ profile }} =~ dev.* ]]; then
+    echo "{{command}}" >&2
+    {{command}}
+  fi
+
+# Runs the provided command if in a pull request
+[private]
+[no-cd]
+in-pr +command:
+  #!/usr/bin/env bash
+  set -euo pipefail
+  if [[ {{ github-event-name }} = pull_request ]]; then
+    echo "{{command}}" >&2
+    {{command}}
+  fi
+
+# Runs the provided command if not in a pull request
+[private]
+[no-cd]
+not-in-pr +command:
+  #!/usr/bin/env bash
+  set -euo pipefail
+  if [[ {{ github-event-name }} != pull_request ]]; then
+    echo "{{command}}" >&2
+    {{command}}
+  fi
+
+
+######################################################################
+## Install scripts
+######################################################################
+
+[private]
+@install-cargo-tool tool install version:
+  `{{tool}} --version | grep -q "{{version}}" || cargo install "{{install}}" --version "{{version}}" --locked --force`
```

Comment:
Does `cargo install` not already do this? If you rerun `cargo install` for the same thing does it override it? (Or is that why you have `--force`?)

Reply:
This covers the case when you have installed the tool manually without cargo. But thinking about this I think maybe we should just rely on `cargo install`? It's at least easier to read and results in the same behavior in 99% of all cases.

Reply:
```
❯ cargo nextest --version
cargo-nextest 0.9.37
❯ cargo install cargo-nextest --version 0.9.37
     Ignored package `cargo-nextest v0.9.37` is already installed, use --force to override
```

Reply:
I'd say I try it out and see if it will re-use the install-action tools. In theory it should.

Reply:
It does not work in CI:
```
error: binary `cargo-hack` already exists in destination
[23](https://github.com/hashintel/hash/actions/runs/4542938790/jobs/8007007610?pr=2292#step:9:24)
Add --force to overwrite
```

Also, it will always output to `stderr` if a binary was not installed as it already exists. I reverted the test.